### ISSUE TITLE
Investigate Keras Model Compression before SQL generation

### ISSUE THREAD

**Issue by antoinecarme** [antoinecarme/keras2sql#5]
This is probably a way to avoid some unnecessarily large models and generate an (almost) equivalent SQL code.

**Comment by antoinecarme** [comment]
There are  at least 4 methods for compressing keras models in 

![image](https://user-images.githubusercontent.com/16481992/42414036-72ef4fbe-822d-11e8-9bc2-6b4e75f69464.png)

https://arxiv.org/pdf/1710.09282.pdf

**Comment by antoinecarme** [comment]
![image](https://user-images.githubusercontent.com/16481992/42414037-834da338-822d-11e8-83fe-1f7b03ff0a0d.png)

**Comment by antoinecarme** [comment]
Would be nice to test this : 

 Model Compression Based on Geoffery Hinton's Logit Regression Method in Keras applied to MNIST 16x compression over 0.95 percent accuracy 

https://github.com/Irtza/Keras_model_compression

**Comment by antoinecarme** [comment]
Factorization method : 

low-rank approximation
    1. with SVD (matrix)
    2. with Tucker (tensor)

https://github.com/DwangoMediaVillage/keras_compressor
